src/main/java/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+package service.impl;
+
+import model.Bike;
+import model.BikeType;
+import model.EBike;
+import model.FoldingBike;
+import model.SpeedElectricBike;
+import service.FileService;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileServiceImpl implements FileService {
+
+    @Override
+    public List<Bike> parseData(List<String[]> data) {
+        List<Bike> list = new ArrayList<>();
+        for (String[] bikeData : data) {
+            String brand = bikeData[0];
+            if (brand.startsWith(BikeType.FOLDING.getBikeTupe())) {
+                FoldingBike foldingBike = FoldingBike.newBuilder()
+                        .setBrand(bikeData[0])
+                        .setNumberOfGears(bikeData[2])
+                        .setAvailabilityOfLights(Boolean.parseBoolean(bikeData[4]))
+                        .setPrice(Integer.parseInt(bikeData[6]))
+                        .build();
+                list.add(foldingBike);
+            } else if (brand.startsWith(BikeType.SPEEDELEC.getBikeTupe())) {
+                SpeedElectricBike speedElectricBike = SpeedElectricBike.newBuilder().
+                        setBrand(bikeData[0])
+                        .setAvailabilityOfLights(Boolean.parseBoolean(bikeData[3]))
+                        .setBatteryCapacity(bikeData[4])
+                        .setPrice(Integer.parseInt(bikeData[6]))
+                        .build();
+                list.add(speedElectricBike);
+            } else {
+                EBike eBike = EBike.newBuilder()
+                        .setBrand(bikeData[0])
+                        .setAvailabilityOfLights(Boolean.parseBoolean(bikeData[3]))
+                        .setBatteryCapacity(bikeData[4])
+                        .setPrice(Integer.parseInt(bikeData[6]))
+                        .build();
+                list.add(eBike);
+            }
+        }
+        return list;
+    }
+
+    @Override
+    public List<String[]> readFile(File file) {
+        List<String[]> list = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] bikeData = new String[7];
+                String[] split = line.split("; ");
+                System.arraycopy(split, 0, bikeData, 0, bikeData.length);
+                list.add(bikeData);
+            }
+        }catch (FileNotFoundException ex){
+            System.out.println("There is no file!");
+        }
+        catch (IOException e) {
+            System.out.println("readFile method " + e);;

```

Comment:
```suggestion
            System.out.println("readFile method " + e);

```
додай трішки більш інформативне повідомлення